services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -205,17 +207,28 @@ public void run()
   {
     try {
       Injector injector = makeInjector();
-
       try {
-        Lifecycle lifecycle = initLifecycle(injector);
-
+        final Lifecycle lifecycle = initLifecycle(injector);
+        Runtime.getRuntime().addShutdownHook(
+            new Thread(
+                new Runnable()
+                {
+                  @Override
+                  public void run()
+                  {
+                    log.info("Running shutdown hook");
+                    lifecycle.stop();
+                  }
+                }
+            )
+        );
         injector.getInstance(ExecutorLifecycle.class).join();
-        lifecycle.stop();
```

Comment:
I'm saying I'm not sure if it's needed or not, it depends on how good the existing tasks are about thread cleanliness on exit. If they're doing things right then the lifecycle.stop() is redundant to the shutdown hook.

@drcrallen suggested taking it out and letting the integration tests catch potential problems, which seems fair.